python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,377 @@
+        doc="Output table to write.",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", ),
+    )
+
+
+class CpMonochromatorScanConfig(pipeBase.PipelineTaskConfig,
+                                pipelineConnections=CpMonochromatorScanConnections):
+    efdClientInstance = pexConfig.Field(
+        dtype=str,
+        doc="EFD instance to use for monochromator results.",
+        default="usdf_efd",
+    )
+    efdUnits = pexConfig.Field(
+        dtype=str,
+        doc="Units to use for EFD monochromator results.",
+        default="nm",
+    )
+    headerDateKey = pexConfig.Field(
+        dtype=str,
+        doc="Header keyword to use for observation date.",
+        default="DATE-BEG",
+    )
+    peakBoxSize = pexConfig.Field(
+        dtype=int,
+        doc="Half-size of the box used for fitting the spectrum peak.",
+        default=50,
+    )
+
+
+class CpMonochromatorScanTask(pipeBase.PipelineTask):
+    """Compare EFD monochromator results to fiber spectrograph spectra.
+    """
+
+    ConfigClass = CpMonochromatorScanConfig
+    _DefaultName = "cpMonochromatorScan"
+
+    def run(self, inputExpHandles):
+        """Match EFD results to spectrograph.
+
+        Parameters
+        ----------
+        inputExpHandles : `list` [`lsst.daf.butler.DeferredDatasetHandle`]
+            Input list of exposure handles to combine.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The results struct containing:
+
+            ``outputData``
+                Final combined filter scan, with a single table
+                containing the commanded monochromator wavelength, as
+                well as the Gaussian fit to the peak of the fiber
+                spectrograph spectrum (`astropy.table.Table`).
+        """
+        monochromatorScanResults = []
+        efdClient = CpEfdClient(efdInstance=self.config.efdClientInstance)
+        monochromatorData = efdClient.getEfdMonochromatorData()
+
+        for handle in inputExpHandles:
+            exposureId = handle.dataId['exposure']
+            spectrum = handle.get()
+            fitMean, fitSigma, peakWavelength, fwhm, fitRange = self._fitPeak(spectrum,
+                                                                              self.config.peakBoxSize)
+            date = spectrum.metadata[self.config.headerDateKey]
+            monoDate, monoValue = efdClient.parseMonochromatorStatus(monochromatorData, date)
+            entry = {
+                'exposure': exposureId,
+                'fit_mean': fitMean,
+                'fit_sigma': fitSigma,
+                'peak': peakWavelength,
+                'fwhm': fwhm,
+                'fit_range': fitRange,
+                'exp_date': date,
+                'mono_date': monoDate.isoformat(),
+                'mono_state': monoValue,
+            }
+            monochromatorScanResults.append(entry)
+
+        # Will this silence the errors?
```

Comment:
I must know!  Will it?

Reply:
The answer is yes, and the comment deleted.